src/core_deku/state.ml

```diff
@@ -9,9 +9,39 @@ type t = {
 }
 [@@deriving yojson]
 
-type receipt = Receipt_tezos_withdraw of Ledger.Withdrawal_handle.t
+type contract_call_changes = {
+  remaining_tickets : (Ticket_id.t * Amount.t) list;
+  additional_operations : additional_operation list;
+  new_storage : bytes;
+}
+
+and additional_operation = {
+  user_operation : User_operation.t;
+  operation_receipt : receipt option;
+}
+
+and contract_origination_changes = {
+  originated_contract_address : Contract_address.t;
+  initial_tickets : (Ticket_id.t * Amount.t) list;
+}
+
+and receipt =
+  | Receipt_tezos_withdraw       of Ledger.Withdrawal_handle.t
+  | Receipt_contract_origination of {
+      sender : Address.t;
+      outcome : [`Success of contract_origination_changes | `Failure];
+    }
+  | Receipt_contract_call        of {
```

Comment:
This should probably be named `Receipt_contract_invocation` in order to keep naming convention. As well as `contract_invocation_changes`.